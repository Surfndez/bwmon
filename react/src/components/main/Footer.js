import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
    footer: {
        margin: theme.spacing(1),
    },
}));

const Footer = () => {
    const classes = useStyles();

    return (
        <Typography variant="subtitle2" className={classes.footer}>
            All usage information is in gigabytes
        </Typography>
    );
};

export default Footer;
