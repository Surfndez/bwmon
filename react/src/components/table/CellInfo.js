export default class CellInfo {
    constructor(id, sortable, align, label, footer, convert) {
        this.id = id;
        this.sortable = sortable;
        this.align = align;
        this.label = label;
        this.footer = footer;
        this.convert = convert;
    }
}
