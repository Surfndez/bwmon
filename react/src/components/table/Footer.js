import React from 'react';
import { TableHead, TableRow, TableCell } from '@material-ui/core';

const Footer = ({ prefix, cellInfos, values }) => (
    <TableHead>
        <TableRow key="footer" id={prefix + '-footer'}>
            {cellInfos.map((cellInfo, index) => {
                const text =
                    index === 0
                        ? 'Totals'
                        : cellInfo.footer
                        ? cellInfo.convert
                            ? cellInfo.convert(values[cellInfo.id])
                            : values[cellInfo.id]
                        : '';
                return (
                    <TableCell key={cellInfo.id} align={cellInfo.align}>
                        {text}
                    </TableCell>
                );
            })}
        </TableRow>
    </TableHead>
);

export default Footer;
