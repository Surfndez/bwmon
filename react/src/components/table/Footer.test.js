import React from 'react';
import { create } from 'react-test-renderer';
import Footer from './Footer';
import CellInfo from './CellInfo';

describe('Footer', () => {
    const convert = (value) => value + 'xyz';

    const cellInfos = [
        new CellInfo('IP', true, 'left', 'IP', true),
        new CellInfo('MAC', false, 'left', 'MAC', false),
        new CellInfo('download', true, 'right', 'Down', true),
        new CellInfo('upload', false, 'right', 'Up', true, convert),
        new CellInfo('total', false, 'right', 'Total', false),
    ];

    const values = {
        IP: 'IP ADDRESS',
        MAC: 'MAC ADDRESS',
        download: 12345,
        upload: 67890,
        total: 2468,
    };

    test('renders correctly', () => {
        const tree = create(<Footer prefix="test" cellInfos={cellInfos} values={values} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
