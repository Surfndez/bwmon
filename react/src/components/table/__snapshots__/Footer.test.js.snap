// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<thead
  className="MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
    id="test-footer"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
      scope="col"
    >
      Totals
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
      scope="col"
    >
      
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      12345
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      67890xyz
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      
    </th>
  </tr>
</thead>
`;
